### PR TITLE
Migrate annot -> key

### PR DESCRIPTION
**Pull request by hchataing**

### CODE REVIEW

File: pdl-compiler/src/parser.rs [hunk 1]
```diff
@@ -191,26 +176,32 @@ pub struct PDLParser;
 
 type Node<'i> = Pair<'i, Rule>;
 type NodeIterator<'i> = Peekable<Filter<Pairs<'i, Rule>, fn(&Node<'i>) -> bool>>;
-type Context<'a> = (crate::ast::FileId, &'a Vec<usize>);
+struct Context<'a>(ast::FileId, &'a Vec<usize>, std::cell::Cell<usize>);
```
Comment (mgeisler) [inline pdl-compiler/src/parser.rs:179]: I saw the `.2` below and wonder if it wouldn't be better to use a struct with named fields here?

**Review by mgeisler — APPROVED** [review]
Hey Henri, this is cool in the way that it removes a lot of type-level shenanigans.

File: pdl-compiler/src/ast.rs [hunk 2]
```diff
@@ -154,10 +149,12 @@ pub enum FieldDesc {
...
 
 #[derive(Debug, Serialize, Clone)]
-pub struct Field<A: Annotation> {
+pub struct Field {
     pub loc: SourceRange,
+    /// Unique identifier used to refer to the AST node in
+    /// compilation environments.
     #[serde(skip_serializing)]
-    pub annot: A::FieldAnnotation,
+    pub key: usize,
```
Comment (DeltaEvo) [inline pdl-compiler/src/ast.rs:157]: nit: Could you create wrapper types to make sure that we don't mix Field and Decl keys ?

File: pdl-compiler/src/parser.rs [hunk 3]
```diff
@@ -191,26 +176,36 @@ pub struct PDLParser;
...
     fn children(self) -> NodeIterator<'i>;
-    fn as_loc(&self, context: &Context) -> crate::ast::SourceRange;
+    fn as_loc(&self, context: &Context) -> ast::SourceRange;
     fn as_string(&self) -> String;
     fn as_usize(&self) -> Result<usize, String>;
 }
 
+impl<'a> Context<'a> {
+    fn key(&self) -> usize {
+        self.key.replace(self.key.get() + 1)
```
Comment (DeltaEvo) [inline pdl-compiler/src/parser.rs:194]:
```suggestion
        self.key.set(self.key.get() + 1)
```

**Review by DeltaEvo — CHANGES_REQUESTED** [review]

File: pdl-compiler/src/ast.rs (hunk 2 above)
Comment (hchataing) [reply pdl-compiler/src/ast.rs]: Done

**Review by DeltaEvo — APPROVED** [review]

**Pull request merged**